account_move_tier_validation_approver/models/res_config_settings.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 ForgeFlow, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    require_approver_in_vendor_bills = fields.Boolean(
+        string="Require Approver In Vendor Bills",
+        help="Requires adding an approver before a vendor bill can be posted.",
+        related="company_id.require_approver_in_vendor_bills",
+        readonly=False,
+    )
+
+    def set_values(self):
+        tier_definition = self.company_id.validation_approver_tier_definition_id
+        if self.require_approver_in_vendor_bills:
+            if not tier_definition and not tier_definition.active:
+                field = self.env["ir.model.fields"].search(
+                    [("model", "=", "account.move"), ("name", "=", "approver_id")]
+                )
+                tier_definition = self.env["tier.definition"].create(
```

Comment:
you need to add a propoer domain as well, otherwise this is affecting customer invoices too.